core/src/main/scala/kafka/security/auth/AzPubSubAclAuthorizer.scala

```diff
@@ -0,0 +1,120 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.security.auth
+
+import com.google.gson.Gson
+import com.typesafe.scalalogging.Logger
+import kafka.network.RequestChannel.Session
+import kafka.utils._
+import org.apache.kafka.common.security.auth.KafkaPrincipal
+import org.apache.kafka.common.security.saml.Token
+
+import scala.util.Try
+import org.apache.kafka.common.utils.{SecurityUtils, Time}
+
+
+class AzPubSubAclAuthorizer extends SimpleAclAuthorizer with Logging {
+  private val authorizerLogger = Logger("kafka.authorizer.logger")
+
+  //
+  // authorizing each request.
+  // if the principal type is SAML token, we need to ensure the token is not expired, despite of the resource type being accessed.
+  // if the request is accessing topic resource, we will do topic authorization; for other type of source,like CLUSTER,
+  // the request should be coming from cluster internal, this kinds of requests should be always allowed.
+  // If the request is coming from channels like PLAINTEXT or SSL, the principal created by Kafka is "User:ANONYMOUS",
+  // we need to make sure in AzPubSubRegistrar (Topics-Prod.ini) the ANONYMOUS user is granted with appropriate permission.
+  //
+  override def authorize(session: Session, operation: Operation, resource: Resource): Boolean = {
```

Comment:
Info req : We may need to support mixed mode during upgrades (Switch from Anonymous -->DSTS)